File: Data Structures & Algorithms/binary-search/submission-6.cs
public class Solution {
    public int Search(int[] nums, int target) {
       int l=0;
       int r=nums.Length;
       while(l<=r){
            int mid = l+(r-l)/2;
            if(mid<0 || mid>=nums.Length) break;
            if(nums[mid]==target) return mid;
            if(nums[mid]<target) l = mid+1;
            if(nums[mid]>target) r = mid-1;
       }
       return -1;
    }
}

File: Data Structures & Algorithms/combination-target-sum/submission-6.java
class Solution {
    public List<List<Integer>> combinationSum(int[] nums, int target) {
        List<List<Integer>> res = new ArrayList<>();
        backtrack(0, nums, 0, target, new ArrayList<Integer>(), res);
        return res;
    }

    public void backtrack(int i, int[] nums, int curSum, int target, List<Integer> ls, List<List<Integer>> res){
        if(curSum==target){
            res.add(new ArrayList(ls));
            return;
        }
        if(i==nums.length ||curSum>target) return;
        ls.add(nums[i]);
        backtrack(i, nums, curSum+nums[i], target, ls, res);
        ls.remove(ls.size()-1);
        backtrack(i+1, nums, curSum, target, ls, res);
    }
}
